op-node/p2p/sync.go

```diff
@@ -668,6 +670,20 @@ func (s *SyncClient) readExecutionPayload(data []byte, expectedTime uint64) (*et
 		return nil, fmt.Errorf("failed to decode response: %w", err)
 	}
 
+	// [Kroma: START] Use KromaDepositTx instead of DepositTx before MPT time
+	if !s.cfg.IsKromaMPT(expectedTime) {
+		for i, otx := range res.Transactions {
+			if otx[0] == types.DepositTxType {
+				var err error
+				res.Transactions[i], err = derive.ToKromaDepositBytes(otx)
+				if err != nil {
+					return nil, fmt.Errorf("failed to convert deposit tx: %w", err)
+				}
+			}
+		}
+	}
+	// [Kroma: END]
+
```

Comment:
Since we already have ecotone applied, shouldn't we consider the case for version 1 ? I don't think there are any p2p nodes requesting version 0.